.github/workflows/CD.yml

```diff
@@ -0,0 +1,119 @@
+name: CD
+
+on:
+  push:
+    tags:        
+      - '**' 
+
+jobs:
+  validate-tag:
+    name: Validate tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-tags: true
+          fetch-depth: 0
+      
+      - name: Check no jumps in tags
+        run: |
+          # The format of a tag (version) should be MAJOR.MINOR
+          # If a previous tag is MAJOR.MINOR, valid tags are only MAJOR.(MINOR+1) or (MAJOR+1).0
+          if [[ $(git tag -l | wc -l) != 1 ]]; then # If there is only one tag, no need to check
+            # Find the previous tag
+            prev=$(git tag -l --sort=-v:refname | head -n 2 | tail -n 1)
+            # Split the previous tag into an array of MAJOR and MINOR version components. E.g., for `3.11` it returns the array (3 11)
+            ver_comp=($(tr "." " " <<< $prev))
+            val1="${ver_comp[0]}.$((${ver_comp[1]}+1))" # MAJOR.(MINOR+1)
+            val2="$((${ver_comp[0]}+1)).0" # (MAJOR+1).0
+            if [[ "${{github.ref_name}}" != "$val1" ]] && \
+            [[ "${{github.ref_name}}" != "$val2" ]]; then
+                echo "Invalid tag \"${{github.ref_name}}\" when previous tag is \"$prev\"."
+                exit 1
+            fi
+          fi
```

Comment:
I see, so we are allowing (for example), going from version `1.0` to version `3.1`?

I am happy to remove this part, but I am wondering why we are only restricting it to model configurations?
Is there any case to purposely want a "jump" in the versioning?

Reply:
> I am happy to remove this part, but I am wondering why we are only restricting it to model configurations?

Because we're using mixed versioning schemes that mean different things: CalVer for models and a modified SemVer for configurations. We're doing this for good reasons, but it is confusing for users. 

The model configuration versioning is tied to reproducibility, so we need to keep this correct and consistent with CI.

The CalVer model scheme needs to be pretty tightly controlled because we're using that for model provenance.

> Is there any case to purposely want a "jump" in the versioning?

No, but in this case we assume developers know what they're doing, and I don't see too much value in constraining it at this point. The number of people who will need to tag these repos is very small. It isn't the same problem space as model configuration and model repositories.

Reply:
Ok, I have removed it in [668885927334c0fcb5064b0f911fff697cf20ca0](https://github.com/ACCESS-NRI/replace_landsurface/pull/45/commits/668885927334c0fcb5064b0f911fff697cf20ca0).